system/Router/AutoRouter.php

```diff
@@ -0,0 +1,286 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Router;
+
+use CodeIgniter\Exceptions\PageNotFoundException;
+
+/**
+ * Router for Auto-Routing
+ */
+class AutoRouter
+{
+    /**
+     * Controller list that can't be accessible.
+     */
+    protected array $protectedControllers;
```

Comment:
It took me quite a time to understand that this array holds CLI routes. At least can we make it more obvious?
```suggestion
    /**
     * List of controllers registered for the CLI verb that should not be accessed in the web.
     */
    protected array $protectedControllers;
```